### ISSUE TITLE
Using null conditional operator crashes Visual Studio

### ISSUE THREAD

**Issue by balajikris** [dotnet/roslyn#8476]
Similar to #8385

1. Get latest code, open a C# project
1. type the follg code:
```C#
 class C
 {
     System.Action a;
     void Foo()
     {
        a?.Invoke();
     }
 }
```
**Result**: Crash when you type `a?.Invoke()`
~~~
System.InvalidOperationException from BoundConditionalReceiver.Accept
Application: devenv.exe
Framework Version: v4.0.30319
Description: The application requested process termination through System.Environment.FailFast(string message).
Message: System.InvalidOperationException: This program location is thought to be unreachable.
   at Microsoft.CodeAnalysis.CSharp.BoundConditionalReceiver.Accept(OperationVisitor visitor)
   at Microsoft.CodeAnalysis.Semantics.OperationWalker.Visit(IOperation operation)
   at Microsoft.CodeAnalysis.Semantics.OperationExtensions.OperationCollector.Visit(IOperation operation)
   at Microsoft.CodeAnalysis.Semantics.OperationWalker.VisitInvocationExpression(IInvocationExpression operation)
   at Microsoft.CodeAnalysis.CSharp.BoundCall.Accept(OperationVisitor visitor)
   at Microsoft.CodeAnalysis.Semantics.OperationWalker.Visit(IOperation operation)
   at Microsoft.CodeAnalysis.Semantics.OperationExtensions.OperationCollector.Visit(IOperation operation)
   at Microsoft.CodeAnalysis.Semantics.OperationWalker.VisitConditionalAccessExpression(IConditionalAccessExpression operation)
   at Microsoft.CodeAnalysis.CSharp.BoundConditionalAccess.Accept(OperationVisitor visitor)
   at Microsoft.CodeAnalysis.Semantics.OperationWalker.Visit(IOperation operation)
   at Microsoft.CodeAnalysis.Semantics.OperationExtensions.OperationCollector.Visit(IOperation operation)
   at Microsoft.CodeAnalysis.Semantics.OperationWalker.VisitExpressionStatement(IExpressionStatement operation)
   at Microsoft.CodeAnalysis.CSharp.BoundExpressionStatement.Accept(OperationVisitor visitor)
   at Microsoft.CodeAnalysis.Semantics.OperationWalker.Visit(IOperation operation)
   at Microsoft.CodeAnalysis.Semantics.OperationExtensions.OperationCollector.Visit(IOperation operation)
   at Microsoft.CodeAnalysis.Semantics.OperationWalker.VisitArray[T](ImmutableArray`1 list)
   at Microsoft.CodeAnalysis.Semantics.OperationWalker.VisitBlockStatement(IBlockStatement operation)
   at Microsoft.CodeAnalysis.CSharp.BoundBlock.Accept(OperationVisitor visitor)
   at Microsoft.CodeAnalysis.Semantics.OperationWalker.Visit(IOperation operation)
   at Microsoft.CodeAnalysis.Semantics.OperationExtensions.OperationCollector.Visit(IOperation operation)
   at Microsoft.CodeAnalysis.Semantics.OperationExtensions.DescendantsAndSelf(IOperation operation)
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerDriver`1.GetOperationsToAnalyze(ImmutableArray`1 operationBlocks)
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerDriver`1.ExecuteDeclaringReferenceActions(SyntaxReference decl, Int32 declarationIndex, SymbolDeclaredCompilationEvent symbolEvent, AnalysisScope analysisScope, AnalysisState analysisStateOpt, Boolean shouldExecuteSyntaxNodeActions, Boolean shouldExecuteOperationActions, Boolean shouldExecuteCodeBlockActions, Boolean shouldExecuteOperationBlockActions, Boolean isInGeneratedCode, CancellationToken cancellationToken)
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerDriver`1.ExecuteDeclaringReferenceActions(SymbolDeclaredCompilationEvent symbolEvent, AnalysisScope analysisScope, AnalysisState analysisStateOpt, Boolean isGeneratedCodeSymbol, CancellationToken cancellationToken)
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerDriver.ProcessSymbolDeclared(SymbolDeclaredCompilationEvent symbolEvent, AnalysisScope analysisScope, AnalysisState analysisStateOpt, CancellationToken cancellationToken)
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerDriver.ProcessEventCore(CompilationEvent e, AnalysisScope analysisScope, AnalysisState analysisStateOpt, CancellationToken cancellationToken)
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerDriver.ProcessEvent(CompilationEvent e, AnalysisScope analysisScope, AnalysisState analysisStateOpt, CancellationToken cancellationToken)
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerDriver.<ProcessCompilationEventsCoreAsync>d__64.MoveNext()
Stack:
   at System.Environment.FailFast(System.String, System.Exception)
   at Microsoft.CodeAnalysis.FailFast.OnFatalException(System.Exception)
   at Microsoft.CodeAnalysis.FatalError.Report(System.Exception, System.Action`1<System.Exception>)
   at Microsoft.CodeAnalysis.FatalError.Report(System.Exception)
   at Microsoft.CodeAnalysis.FatalError.ReportUnlessCanceled(System.Exception)
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerDriver+<ProcessCompilationEventsCoreAsync>d__64.MoveNext()
   at Microsoft.CodeAnalysis.CSharp.BoundConditionalReceiver.Accept(Microsoft.CodeAnalysis.Semantics.OperationVisitor)
   at Microsoft.CodeAnalysis.Semantics.OperationWalker.Visit(Microsoft.CodeAnalysis.Semantics.IOperation)
   at Microsoft.CodeAnalysis.Semantics.OperationExtensions+OperationCollector.Visit(Microsoft.CodeAnalysis.Semantics.IOperation)
   at Microsoft.CodeAnalysis.Semantics.OperationWalker.VisitInvocationExpression(Microsoft.CodeAnalysis.Semantics.IInvocationExpression)
   at Microsoft.CodeAnalysis.CSharp.BoundCall.Accept(Microsoft.CodeAnalysis.Semantics.OperationVisitor)
   at Microsoft.CodeAnalysis.Semantics.OperationWalker.Visit(Microsoft.CodeAnalysis.Semantics.IOperation)
   at Microsoft.CodeAnalysis.Semantics.OperationExtensions+OperationCollector.Visit(Microsoft.CodeAnalysis.Semantics.IOperation)
   at Microsoft.CodeAnalysis.Semantics.OperationWalker.VisitConditionalAccessExpression(Microsoft.CodeAnalysis.Semantics.IConditionalAccessExpression)
   at Microsoft.CodeAnalysis.CSharp.BoundConditionalAccess.Accept(Microsoft.CodeAnalysis.Semantics.OperationVisitor)
   at Microsoft.CodeAnalysis.Semantics.OperationWalker.Visit(Microsoft.CodeAnalysis.Semantics.IOperation)
   at Microsoft.CodeAnalysis.Semantics.OperationExtensions+OperationCollector.Visit(Microsoft.CodeAnalysis.Semantics.IOperation)
   at Microsoft.CodeAnalysis.Semantics.OperationWalker.VisitExpressionStatement(Microsoft.CodeAnalysis.Semantics.IExpressionStatement)
   at Microsoft.CodeAnalysis.CSharp.BoundExpressionStatement.Accept(Microsoft.CodeAnalysis.Semantics.OperationVisitor)
   at Microsoft.CodeAnalysis.Semantics.OperationWalker.Visit(Microsoft.CodeAnalysis.Semantics.IOperation)
   at Microsoft.CodeAnalysis.Semantics.OperationExtensions+OperationCollector.Visit(Microsoft.CodeAnalysis.Semantics.IOperation)
   at Microsoft.CodeAnalysis.Semantics.OperationWalker.VisitArray[[System.__Canon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]](System.Collections.Immutable.ImmutableArray`1<System.__Canon>)
   at Microsoft.CodeAnalysis.Semantics.OperationWalker.VisitBlockStatement(Microsoft.CodeAnalysis.Semantics.IBlockStatement)
   at Microsoft.CodeAnalysis.CSharp.BoundBlock.Accept(Microsoft.CodeAnalysis.Semantics.OperationVisitor)
   at Microsoft.CodeAnalysis.Semantics.OperationWalker.Visit(Microsoft.CodeAnalysis.Semantics.IOperation)
   at Microsoft.CodeAnalysis.Semantics.OperationExtensions+OperationCollector.Visit(Microsoft.CodeAnalysis.Semantics.IOperation)
   at Microsoft.CodeAnalysis.Semantics.OperationExtensions.DescendantsAndSelf(Microsoft.CodeAnalysis.Semantics.IOperation)
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerDriver`1[[Microsoft.CodeAnalysis.CSharp.SyntaxKind, Microsoft.CodeAnalysis.CSharp, Version=42.42.42.42, Culture=neutral, PublicKeyToken=31bf3856ad364e35]].GetOperationsToAnalyze(System.Collections.Immutable.ImmutableArray`1<Microsoft.CodeAnalysis.Semantics.IOperation>)
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerDriver`1[[Microsoft.CodeAnalysis.CSharp.SyntaxKind, Microsoft.CodeAnalysis.CSharp, Version=42.42.42.42, Culture=neutral, PublicKeyToken=31bf3856ad364e35]].ExecuteDeclaringReferenceActions(Microsoft.CodeAnalysis.SyntaxReference, Int32, Microsoft.CodeAnalysis.Diagnostics.SymbolDeclaredCompilationEvent, Microsoft.CodeAnalysis.Diagnostics.AnalysisScope, Microsoft.CodeAnalysis.Diagnostics.AnalysisState, Boolean, Boolean, Boolean, Boolean, Boolean, System.Threading.CancellationToken)
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerDriver`1[[Microsoft.CodeAnalysis.CSharp.SyntaxKind, Microsoft.CodeAnalysis.CSharp, Version=42.42.42.42, Culture=neutral, PublicKeyToken=31bf3856ad364e35]].ExecuteDeclaringReferenceActions(Microsoft.CodeAnalysis.Diagnostics.SymbolDeclaredCompilationEvent, Microsoft.CodeAnalysis.Diagnostics.AnalysisScope, Microsoft.CodeAnalysis.Diagnostics.AnalysisState, Boolean, System.Threading.CancellationToken)
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerDriver.ProcessSymbolDeclared(Microsoft.CodeAnalysis.Diagnostics.SymbolDeclaredCompilationEvent, Microsoft.CodeAnalysis.Diagnostics.AnalysisScope, Microsoft.CodeAnalysis.Diagnostics.AnalysisState, System.Threading.CancellationToken)
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerDriver.ProcessEventCore(Microsoft.CodeAnalysis.Diagnostics.CompilationEvent, Microsoft.CodeAnalysis.Diagnostics.AnalysisScope, Microsoft.CodeAnalysis.Diagnostics.AnalysisState, System.Threading.CancellationToken)
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerDriver.ProcessEvent(Microsoft.CodeAnalysis.Diagnostics.CompilationEvent, Microsoft.CodeAnalysis.Diagnostics.AnalysisScope, Microsoft.CodeAnalysis.Diagnostics.AnalysisState, System.Threading.CancellationToken)
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerDriver+<ProcessCompilationEventsCoreAsync>d__64.MoveNext()
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1[[System.__Canon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]].Start[[System.__Canon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]](System.__Canon ByRef)
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerDriver.ProcessCompilationEventsCoreAsync(Microsoft.CodeAnalysis.Diagnostics.AnalysisScope, Microsoft.CodeAnalysis.Diagnostics.AnalysisState, Boolean, System.Threading.CancellationToken)
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerDriver+<ProcessCompilationEventsAsync>d__63.MoveNext()
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder.Start[[System.__Canon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]](System.__Canon ByRef)
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerDriver.ProcessCompilationEventsAsync(Microsoft.CodeAnalysis.Diagnostics.AnalysisScope, Microsoft.CodeAnalysis.Diagnostics.AnalysisState, Boolean, System.Threading.CancellationToken)
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerDriver+<ExecutePrimaryAnalysisTaskAsync>d__42.MoveNext()
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder.Start[[System.__Canon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]](System.__Canon ByRef)
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerDriver.ExecutePrimaryAnalysisTaskAsync(Microsoft.CodeAnalysis.Diagnostics.AnalysisScope, Microsoft.CodeAnalysis.Diagnostics.AnalysisState, Boolean, System.Threading.CancellationToken)
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerDriver+<AttachQueueAndProcessAllEventsAsync>d__40.MoveNext()
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder.Start[[System.__Canon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]](System.__Canon ByRef)
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerDriver.AttachQueueAndProcessAllEventsAsync(Microsoft.CodeAnalysis.Diagnostics.AsyncQueue`1<Microsoft.CodeAnalysis.Diagnostics.CompilationEvent>, Microsoft.CodeAnalysis.Diagnostics.AnalysisScope, Microsoft.CodeAnalysis.Diagnostics.AnalysisState, System.Threading.CancellationToken)
   at Microsoft.CodeAnalysis.Diagnostics.CompilationWithAnalyzers+<ComputeAnalyzerDiagnosticsCoreAsync>d__55.MoveNext()
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder.Start[[System.__Canon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]](System.__Canon ByRef)
   at Microsoft.CodeAnalysis.Diagnostics.CompilationWithAnalyzers.ComputeAnalyzerDiagnosticsCoreAsync(Microsoft.CodeAnalysis.Diagnostics.AnalyzerDriver, Microsoft.CodeAnalysis.Diagnostics.AsyncQueue`1<Microsoft.CodeAnalysis.Diagnostics.CompilationEvent>, Microsoft.CodeAnalysis.Diagnostics.AnalysisScope, System.Threading.CancellationToken)
   at Microsoft.CodeAnalysis.Diagnostics.CompilationWithAnalyzers+<>c__DisplayClass52_2+<<ComputeAnalyzerDiagnosticsAsync>b__1>d.MoveNext()
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder.Start[[System.__Canon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]](System.__Canon ByRef)
   at Microsoft.CodeAnalysis.Diagnostics.CompilationWithAnalyzers+<>c__DisplayClass52_2.<ComputeAnalyzerDiagnosticsAsync>b__1()
   at System.Threading.Tasks.Task`1[[System.__Canon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]].InnerInvoke()
   at System.Threading.Tasks.Task.Execute()
   at System.Threading.Tasks.Task.ExecutionContextCallback(System.Object)
   at System.Threading.ExecutionContext.RunInternal(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean)
   at System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean)
   at System.Threading.Tasks.Task.ExecuteWithThreadLocal(System.Threading.Tasks.Task ByRef)
   at System.Threading.Tasks.Task.ExecuteEntry(Boolean)
   at System.Threading.Tasks.Task.System.Threading.IThreadPoolWorkItem.ExecuteWorkItem()
   at System.Threading.ThreadPoolWorkQueue.Dispatch()
   at System.Threading._ThreadPoolWaitCallback.PerformWaitCallback()
~~~

**Comment by balajikris** [comment]
cc: @genlu @srivatsn

**Comment by genlu** [comment]
Thanks for filing this @balajikris 
@JohnHamby @srivatsn Any suggestion on how to represent this as an operation? Simply expanding 
`foo?.Bar` => `foo == null ? null : foo.Bar`?

**Comment by genlu** [comment]
Here's a more comprehensive test case for this:
```cs
class C
{
    public int Prop { get; set; }

    public int Field;

    public int this[int i]
    {
        get
        {
            return this.Field;
        }
        set
        {
            this.Field = value;
        }
    }

    public void M0(C foo)
    {
        var x = foo?.Prop;
        x = foo?.Field;
        x = foo?[0];
        foo?.M0(null);
    }
}
```

Note that the expression `foo?[0]` is a `BoundIndexerAccess` which is also an `OperationKind.None` kind. This needs to be fixed as well.

FYI @JohnHamby

**Comment by genlu** [comment]
Also I have verified this crash can be avoided by #8517, but it is not a fix for the bug. A proper fix will come soon.

**Comment by genlu** [comment]
Another problem exposed by this issue is in `IConditionalAccessExpression`, only an `Access` expression is provided, whereas the receiver is not. We need to change the interface to include it.

**Comment by genlu** [comment]
Removed the `Urgency-now` label since the crash is fixed by #8517 .

**Comment by JohnHamby** [comment]
Resolved by #8669 .
